contracts/utils/cryptography/ECDSA.sol

```diff
@@ -180,10 +180,14 @@ library ECDSA {
      *
      * See {recover}.
      */
-    function toEthSignedMessageHash(bytes32 hash) internal pure returns (bytes32) {
-        // 32 is the length in bytes of hash,
+    function toEthSignedMessageHash(bytes32 hash) internal pure returns (bytes32 message) {
         // enforced by the type signature above
-        return keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hash));
+        /// @solidity memory-safe-assembly
+        assembly {
+            mstore(0, "\x19Ethereum Signed Message:\n32")
+            mstore(28, hash)
+            message := keccak256(0, 60)
+        }
```

Comment:
I believe using hex values is more clear that this doesn not overwrite reserved memory as `0x40`

```suggestion
        assembly {
            mstore(0x00, "\x19Ethereum Signed Message:\n32")
            mstore(0x1C, hash)
            message := keccak256(0x00, 0x3C)
        }
```